draft-wood-key-consistency.md

```diff
@@ -148,55 +158,132 @@ to ensure consistency here might include, though are not limited to:
-impractical. In larger scale deployments, the short-comings of this model may be similar to {{log-based}}.
-
-If the database is small and users query it infrequently, retrieval techniques based on PIR may be viable.
-
-## Minimum Validity Periods
+## Database Discovery {#external-db-based}
+
+In this model, servers publish keys in an external database and clients fetch keys from the database, as shown below.
+
+~~~
+┌──────────┐
+│          │
+│  Client  ├───────────┐
+│          │           │
+└──────────┘           │
+                       │
+┌──────────┐         ┌─▼────────┐      ┌─────────┐
+│          │         │          │      │         │
+│  Client  ├─────────► Database ◄──────┤ Server  │
+│          │         │          │      │         │
+└──────────┘         └─▲────────┘      └─────────┘
+     x                 │
+     x                 │
+     x                 │
+┌──────────┐           │
+│          │           │
+│  Client  ├───────────┘
+│          │
+└──────────┘
+~~~
+{: #fig-disc-database title="Database Discovery Example"}
+
+The database is expected to have a table that asserts mappings between server names and keys. Examples
+of such databases are as follows:
+
+- An append-only, audited table similar to that of Certificate Transparency {{!RFC6962}}. The log is operated and
+  audited in such a way that the contents of the log are consistent for all users. Any reliant system
+  which depends on this type of KCCS requires the log be audited or users have some other mechanism for
+  checking their view of the log state (gossiping). However, this type of system does not ensure proactive
+  security against malicious servers unless log participants actively check log proofs. This requirement
+  may impede deployment in practice, given that no web browser checks SignedCertificateTimestamps before
+  using (accepting as valid) a corresponding TLS certificate.
+
+- A consensus-based table whose assertions are created by a coalition of entities that periodically agree on
+  the correct binding of server names and key material. In this model the agreement is achieved via a consensus
+  protocol, but the specific consensus protocol is dependent on the implementation. For privacy, users should
+  either download the entire database and query it locally, or remotely query the database using a private
+  information retrieval (PIR) protocol. In the case where the database is downloaded locally, it should be
+  considered stale and re-fetch periodically, as well.
+
+  When the entire database is downloaded, this model is appropriate in small scale deployments where the
+  number of bindings in the database is much smaller than the number of users of the reliant system. In
+  a reliant system with a large user base, this model imposes bandwidth costs on each user that may be
+  impractical. In larger scale deployments, the short-comings of this model may be similar to log-based
```

Comment:
I'm not sure that having a large number of users is a particular problem here based on our experience with CRLite and other such things.  As the update period can be relatively long, the cost of periodic updates should be manageable.  At most, you are talking about key record updates that occur on a similar period to software update checks (assuming reasonable practices there also).

Reply:
👍  I removed this bit.